core/sdk-core/src/main/java/software/amazon/awssdk/core/client/builder/SdkDefaultClientBuilder.java

```diff
@@ -222,6 +223,7 @@ private SdkClientConfiguration setOverrides(SdkClientConfiguration configuration
 
         SdkClientConfiguration.Builder builder = configuration.toBuilder();
 
+        builder.option(SCHEDULED_EXECUTOR_SERVICE, clientOverrideConfiguration.scheduledExecutorService().orElse(null));
```

Comment:
Do we need to set it here since it's going to be resolved and overriden in `finalizeConfiguration`?

Reply:
Seems to me the pattern is to push every params in the attributes map of `SdkClientConfiguration` and finalize with only this attributes map as input/output. So `clientOverrideConfiguration` will never be read after this. Thus, it is needed.

Reply:
Ah yeah, I read it wrong.